### PR TITLE
Adds neck-snapping to CQC and Sleeping carp

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request
If you achieve a kill-grab on your target as a sleeping carp or CQC user, you can use harm on their head to snap their neck, killing them instantly.
## Why It's Good For The Game
Neck snapping is in line with that secret-agent/commando badassery that sleeping carp and CQC emulate. If you are in a kill grab from a carp or CQC user, you were probably dead anyways, so this shouldn't have large balance implications.
## Changelog
:cl: itseasytosee
add: Sleeping carp/cqc users can now snap peoples necks by punching them in the head while they are in a kill grab.
/:cl:

### CODE REVIEW

File: code/datums/martial/cqc.dm [hunk 1]
```diff
@@ -183,6 +183,17 @@
...
+	if((attacker.grab_state == GRAB_KILL) && attacker.zone_selected == BODY_ZONE_HEAD && defender.stat != DEAD)
+		var/obj/item/bodypart/head = defender.get_bodypart("head")
+		if(head)
+			playsound(defender, 'sound/effects/wounds/crack1.ogg', 160)
+			defender.visible_message(span_danger("[attacker] snaps the neck of [defender]!"), \
+						span_userdanger("Your neck is snapped by [attacker]!"), span_hear("You hear a sickening snap!"), null, attacker)
+			to_chat(attacker, span_danger("In a swift motion, you snap the neck of [defender]!"))
+			log_combat(attacker, defender, "snapped neck")
+			defender.apply_damage(100, BRUTE, BODY_ZONE_HEAD, wound_bonus=CANT_WOUND)
+			defender.death()
```
Comment (MrMelbert) [inline code/datums/martial/cqc.dm:195]: This bypasses `TRAIT_NODEATH`

**Review by MrMelbert — COMMENTED** [review]
Seems kinda OP since CQC gets instant aggro grabs, also if you don't know they're a martial arts user and they instantly kill you

**Comment by SkeletalElite** [comment]
> Seems kinda OP since CQC gets instant aggro grabs, also if you don't know they're a martial arts user and they instantly kill you

Once you get to this grab state you are literally unable to escape without outside intervention and the person grabbing you has punches that do very large amount of damage. At best it will let you kill someone a few seconds sooner than you otherwise would have by just spamming punche on the dude you have grabbed instead of sitting there upgrading your grab an extra time

**Comment by Kapu1178** [comment]
this doesn't feel very in theme with Sleeping Carp imo

**Comment by necromanceranne** [comment]
> this doesn't feel very in theme with Sleeping Carp imo

Well, unfortunately, you're incorrect.

**Comment by itseasytosee** [comment]
> Seems kinda OP since CQC gets instant aggro grabs, also if you don't know they're a martial arts user and they instantly kill you

Sleeping carp can already get a 40 second sleep by disarming with an aggressive grab, the truth is the game was rigged from the start.

**Comment by SomeoneYouProbablyKnow** [comment]
The question is not powerlevel but the fact that this is lame. Yes it's only marginally more lame than current insta-kill things cqc can do but something being bad is not a good reason to make it worse.

At least as it is you can't insta kill someone with cqc in putched combat, you can take em out but they'll get back up if you don't finish them. This just lets you kill people with a few clicks.

**Comment by DaCoolBoss** [comment]
I'd like it if it took more than one attempt. Not for balance reasons, I just think it's cooler to have the player mash inputs a few times like in metal gear solid. It's something I've missed since quick grabs were patched out.

**Comment by necromanceranne** [comment]
Both of these martial arts will put you into a state of 'effectively dead' fairly rapidly. I don't see how someone who took the time to put you into a kill grab, which takes bloody forever since you have to elevate above a neckgrab to do it, couldn't just kick you to death instead.

Especially sleeping carp, they can behead you long before this even matters.

**Comment by Dorsisdwarf** [comment]
Well it definitely makes a difference if the CQCer is fighting more than one person, since the # 1 group of people sleeping carp users are likely to get into an actual fight with (crew and sec) tend to move in packs, and "straight to dead" is a HUGE improvement over "knocked out" in a running group combat.

**Comment by moocowswag** [comment]
So the point of this is to counter armor/augs/regeneration to basically make the time to kill someone flat regardless of what they have unless they are a hulk?

**Comment by ATH1909** [comment]
> Well it definitely makes a difference if the CQCer is fighting more than one person, since the # 1 group of people sleeping carp users are likely to get into an actual fight with (crew and sec) tend to move in packs, and "straight to dead" is a HUGE improvement over "knocked out" in a running group combat.

but if you're in a group combat, the other enemies won't let you finish the two do_after()s needed to escalate to a kill grab

lowkey the requirement here could probably be lowered to a neck grab, needing two do_after()s (of moderate length) is kind of awkward for a simple neck snap

File: code/datums/martial/cqc.dm (hunk 1 above)
Comment (MrMelbert) [reply code/datums/martial/cqc.dm:195]:
```suggestion
			if(!HAS_TRAIT(defender, TRAIT_NODEATH))
				defender.death()
```

**Comment by SkeletalElite** [comment]
> Well it definitely makes a difference if the CQCer is fighting more than one person, since the # 1 group of people sleeping carp users are likely to get into an actual fight with (crew and sec) tend to move in packs, and "straight to dead" is a HUGE improvement over "knocked out" in a running group combat.

you have to upgrade your grab three times though, which takes forever. Someone with sleeping carp can punch you into crit faster than that. "straight to dead" will never happen in a group fight because passive grab > aggro grab > neck grab > kill grab takes way too long to be viable in a group fight and the neck grab  and kill grab upgrades require you stand totally still

**Comment by deathrobotpunch** [comment]
well sleeping carp and qcq is more meant for stealth kills anyways

File: code/datums/martial/cqc.dm [hunk 2]
```diff
@@ -183,6 +183,20 @@
 		log_combat(attacker, defender, "sweeped (CQC)")
 		reset_streak()
 		return TRUE
+	if((attacker.grab_state == GRAB_KILL) && attacker.zone_selected == BODY_ZONE_HEAD && defender.stat != DEAD)
+		var/obj/item/bodypart/head = defender.get_bodypart("head")
+		if(head)
+			playsound(defender, 'sound/effects/wounds/crack1.ogg', 160)
```
Comment (Fikou) [inline code/datums/martial/cqc.dm:189]: thats too loud

File: code/datums/martial/sleeping_carp.dm [hunk 3]
```diff
@@ -107,6 +107,17 @@
 	return ..()
 
 /datum/martial_art/the_sleeping_carp/harm_act(mob/living/attacker, mob/living/defender)
+	if((attacker.grab_state == GRAB_KILL) && attacker.zone_selected == BODY_ZONE_HEAD && defender.stat != DEAD)
+		var/obj/item/bodypart/head = defender.get_bodypart("head")
+		if(head)
+			playsound(defender, 'sound/effects/wounds/crack1.ogg', 160)
+			defender.visible_message(span_danger("[attacker] snaps the neck of [defender]!"), \
+						span_userdanger("Your neck is snapped by [attacker]!"), span_hear("You hear a sickening snap!"), null, attacker)
```
Comment (Mothblocks) [inline code/datums/martial/sleeping_carp.dm:115]:
```suggestion
			defender.visible_message(
				span_danger("[attacker] snaps the neck of [defender]!"),
				span_userdanger("Your neck is snapped by [attacker]!"),
				span_hear("You hear a sickening snap!"),
				replace_with_named_parameter_of_whatever_this_is = attacker,
			)
```

Then do this iwth the other one

File: code/datums/martial/sleeping_carp.dm [hunk 4]
```diff
@@ -107,6 +107,17 @@
 	return ..()
 
 /datum/martial_art/the_sleeping_carp/harm_act(mob/living/attacker, mob/living/defender)
+	if((attacker.grab_state == GRAB_KILL) && attacker.zone_selected == BODY_ZONE_HEAD && defender.stat != DEAD)
+		var/obj/item/bodypart/head = defender.get_bodypart("head")
+		if(head)
+			playsound(defender, 'sound/effects/wounds/crack1.ogg', 160)
```
Comment (Mothblocks) [inline code/datums/martial/sleeping_carp.dm:113]: 160 is so fucking loud bro

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/datums/martial/sleeping_carp.dm (hunk 4 above)
Comment (DrDiasyl) [reply code/datums/martial/sleeping_carp.dm]: > 160 is so fucking loud bro

Neck bones are loud

**Comment by chemistrymain2** [comment]
balance it by giving you a +2 moodlet that tells you that they finally got that awful crick out of your neck and it'll be fine

high intensity neck massage

**Comment by itseasytosee** [comment]
Johnathon Gupta Said this PR must not be closed so take that as you will maintainers.

**Pull request merged**